app/components/avo/actions_component.rb

```diff
@@ -2,40 +2,47 @@
 
 class Avo::ActionsComponent < Avo::BaseComponent
   include Avo::ApplicationHelper
-  attr_reader :label, :size, :as_row_control
-
-  def initialize(actions: [], resource: nil, view: nil, exclude: [], include: [], style: :outline, color: :primary, label: nil, size: :md, as_row_control: false, icon: nil)
-    @actions = actions || []
-    @resource = resource
-    @view = view
-    @exclude = Array(exclude)
-    @include = include
-    @color = color
-    @style = style
-    @label = label || I18n.t("avo.actions")
-    @size = size
-    @icon = icon
-    @as_row_control = as_row_control
+
+  ACTION_FILTER = _Set(_Class(Avo::BaseAction))
+
+  prop :actions, _Array(Avo::BaseAction), default: -> { [] }
+  prop :resource, _Nilable(Avo::BaseResource)
+  prop :view, _Nilable(Symbol), &:to_sym
+  prop :exclude, _Nilable(ACTION_FILTER), &:to_set
+  prop :include, _Nilable(ACTION_FILTER), &:to_set
+  prop :style, Avo::ButtonComponent::STYLE, default: :outline
+  prop :color, Symbol, default: :primary
+  prop :label, String do |value|
+    value || I18n.t("avo.actions")
+  end
```

Comment:
As far as I can tell, this string should be non-nilable because we’ll always get a string back from the I18n lookup.

Reply:
yes, can't we use the default option with the i18n string?